plugins/funind/g_indfun.mlg

```diff
@@ -176,8 +176,8 @@ let () =
 }
 
 (* TASSI: n'importe quoi ! *)
-VERNAC COMMAND EXTEND Function
-| ![ proof ] ["Function" ne_function_rec_definition_loc_list_sep(recsl,"with")]
+VERNAC COMMAND EXTEND Function STATE maybe_open_proof
+| ["Function" ne_function_rec_definition_loc_list_sep(recsl,"with")]
```

Comment:
I had a minute to look at the classification function of `Function`.
`Function` extends `Fixpoint` (by reusing the same argument ne_function_rec_definition_loc_list_sep``, that I guess was not factored when I wrote the comment above) with an annotation `{measure}` and `{wf}`. Exactly sa `Fixpoint` one can put a `:= term.` or not after the type. If you do put the term the you don't open a proof *unless* you gave one of the two annotations that are specific to `Function`.

The code in the `=> {...}` STM classification calls the classifier on the AST of `Fixpoint` just to know what it says, and then patches the result if needed.

To me, *statically*, one knows if `Function` opens a proof or not. The problem is that two commands with different types according to your classification are attached to the same grammar entry. Instead of inventing a new `maybe` type, please refactor the code so that there are two entry points, with different types.

Reply:
>To me, statically, one knows if Function opens a proof or not. The problem is that two commands with different types according to your classification are attached to the same grammar entry. Instead of inventing a new maybe type, please refactor the code so that there are two entry points, with different types.

I'm not doing that in this PR, it's complicated enough already.

Reply:
I'll give it a try then